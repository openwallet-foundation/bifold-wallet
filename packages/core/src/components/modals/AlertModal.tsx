import React from 'react'
import { useTranslation } from 'react-i18next'
import { StyleSheet, useWindowDimensions } from 'react-native'
import { SafeAreaView } from 'react-native-safe-area-context'

import { InfoBoxType } from '../misc/InfoBox'

import PopupModal from './PopupModal'

interface AlertModalProps {
  title: string
  message: string
  submit?: () => void
}

const AlertModal: React.FC<AlertModalProps> = ({ title, message, submit }) => {
  const { t } = useTranslation()
  const { height } = useWindowDimensions()

  const styles = StyleSheet.create({
    container: {
      minHeight: height,
      flexDirection: 'column',
      justifyContent: 'center',
      alignItems: 'center',
    },
  })

  return (
    <SafeAreaView style={styles.container}>
      <PopupModal
        notificationType={InfoBoxType.Info}
        title={title}
        description={message}
        onCallToActionLabel={t('Global.Okay')}
        onCallToActionPressed={submit ? () => submit() : () => undefined}
      ></PopupModal>
    </SafeAreaView>
  )
}

export default AlertModal
