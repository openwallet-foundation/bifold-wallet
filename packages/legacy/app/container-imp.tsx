import { BifoldLogger, Container, TokenMapping } from '@hyperledger/aries-bifold-core'
import { DependencyContainer } from 'tsyringe'

export class AppContainer implements Container {
  private _container: DependencyContainer
  private log?: BifoldLogger

  public constructor(bifoldContainer: Container, log?: BifoldLogger) {
    this._container = bifoldContainer.container.createChildContainer()
    this.log = log
  }

  public get container(): DependencyContainer {
    return this._container
  }

  public init(): Container {
    // eslint-disable-next-line no-console
    this.log?.info(`Initializing App container`)
    // Here you can register any component to override components in core package
    // Example: Replacing button in core with custom button
    // this.container.registerInstance(TOKENS.COMP_BUTTON, Button)

    //This is an example of how to customize the screen layout and use custom header for wallets who wnat to hide default navigation header
    //To hide navigation header for a specific page, use headerShown: false in the screen options like this
    /*
    this.container.registerInstance(TOKENS.OBJECT_SCREEN_CONFIG, {
      ...DefaultScreenOptionsDictionary,
      [Screens.Terms]: {
        ...DefaultScreenOptionsDictionary[Screens.Terms],
        headerShown: false,
      },
    })

    //Customizing Terms screen custom header
    this.container.registerInstance(TOKENS.OBJECT_LAYOUT_CONFIG, {
      ...DefaultScreenLayoutOptions,
      [Screens.Terms]: {
        ...DefaultScreenLayoutOptions[Screens.Terms],
        customEdges: ['bottom'],
        safeArea: true,
        Header: () => (
          <View style={{ backgroundColor: 'red', height: 129, justifyContent: 'center', alignItems: 'center' }}>
            <Text style={{ color: 'white' }}>Custom Header</Text>
          </View>
        ),
      },
    })
      */
    return this
  }

  public resolve<K extends keyof TokenMapping>(token: K): TokenMapping[K] {
    return this._container.resolve(token) as TokenMapping[K]
  }
  public resolveAll<K extends keyof TokenMapping, T extends K[]>(
    tokens: [...T]
  ): { [I in keyof T]: TokenMapping[T[I]] } {
    return tokens.map((key) => this.resolve(key)!) as { [I in keyof T]: TokenMapping[T[I]] }
  }
}
