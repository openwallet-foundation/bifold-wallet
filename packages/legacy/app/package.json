{
  "name": "aries-bifold-app",
  "version": "1.0.0",
  "private": true,
  "description": "Aries Bifold Wallet App",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/openwallet-foundation/bifold-wallet.git",
    "directory": "packages/legacy/app"
  },
  "homepage": "https://github.com/openwallet-foundation/bifold-wallet/",
  "scripts": {
    "android": "react-native run-android",
    "gradle": "./android/gradlew --project-dir=./android --no-daemon",
    "gradle:write-locks": "cd android && ./gradlew app:dependencies --write-locks",
    "prepare": "patch-package",
    "ios:setup": "bundle install --frozen && cd ios && bundle exec pod install",
    "ios": "react-native run-ios",
    "start": "react-native start",
    "test": "echo 'nothing to do'"
  },
  "dependencies": {
    "@credo-ts/anoncreds": "0.5.11",
    "@credo-ts/askar": "0.5.11",
    "@credo-ts/core": "0.5.11",
    "@credo-ts/indy-sdk-to-askar-migration": "0.5.11",
    "@credo-ts/indy-vdr": "0.5.11",
    "@credo-ts/openid4vc": "0.5.11",
    "@credo-ts/push-notifications": "^0.7.0",
    "@credo-ts/question-answer": "0.5.11",
    "@credo-ts/react-hooks": "^0.6.0",
    "@credo-ts/react-native": "0.5.11",
    "@formatjs/intl-datetimeformat": "4.2.4",
    "@formatjs/intl-displaynames": "5.2.4",
    "@formatjs/intl-getcanonicallocales": "1.7.3",
    "@formatjs/intl-listformat": "6.3.4",
    "@formatjs/intl-locale": "2.4.38",
    "@formatjs/intl-numberformat": "7.2.4",
    "@formatjs/intl-pluralrules": "4.1.4",
    "@formatjs/intl-relativetimeformat": "9.3.1",
    "@hyperledger/anoncreds-react-native": "0.2.4",
    "@hyperledger/aries-askar-react-native": "0.2.3",
    "@hyperledger/aries-bifold-core": "^1.0.0",
    "@hyperledger/aries-oca": "^1.0.0",
    "@hyperledger/indy-vdr-react-native": "0.2.2",
    "@react-native-async-storage/async-storage": "1.15.11",
    "@react-native-community/masked-view": "0.1.11",
    "@react-native-community/netinfo": "^9.3.0",
    "@react-navigation/bottom-tabs": "6.0.9",
    "@react-navigation/core": "6.1.0",
    "@react-navigation/devtools": "^6.0.8",
    "@react-navigation/native": "6.0.6",
    "@react-navigation/stack": "6.0.11",
    "axios": "^1.4.0",
    "base-64": "^1.0.0",
    "i18next": "^21.4.0",
    "lodash.flatten": "^4.4.0",
    "lodash.startcase": "^4.4.0",
    "moment": "^2.29.4",
    "patch-package": "^6.4.4",
    "query-string": "^7.0.1",
    "react": "*",
    "react-i18next": "11.13.0",
    "react-native": "*",
    "react-native-animated-pagination-dots": "^0.1.72",
    "react-native-argon2": "^2.0.1",
    "react-native-bouncy-checkbox": "^3.0.5",
    "react-native-collapsible": "^1.6.1",
    "react-native-config": "^1.4.2",
    "react-native-confirmation-code-field": "^7.3.0",
    "react-native-device-info": "^8.4.9",
    "react-native-encrypted-storage": "^4.0.2",
    "react-native-fs": "^2.16.6",
    "react-native-gesture-handler": "^1.10.3",
    "react-native-get-random-values": "^1.7.0",
    "react-native-gifted-chat": "2.4.0",
    "react-native-keychain": "^8.1.1",
    "react-native-localize": "^2.2.4",
    "react-native-logs": "^5.1.0",
    "react-native-orientation-locker": "^1.6.0",
    "react-native-permissions": "^4.0.1",
    "react-native-qrcode-svg": "6.2.0",
    "react-native-reanimated": "3.4.2",
    "react-native-safe-area-context": "^3.2.0",
    "react-native-scalable-image": "^1.1.0",
    "react-native-screens": "^3.24.0",
    "react-native-splash-screen": "^3.3.0",
    "react-native-svg": "^12.1.1",
    "react-native-tcp-socket": "^6.0.6",
    "react-native-toast-message": "^2.1.6",
    "react-native-uuid": "^2.0.1",
    "react-native-vector-icons": "^10.0.0",
    "react-native-vision-camera": "4.3.1",
    "react-native-webview": "^11.22.3",
    "tsyringe": "^4.7.0",
    "uuid": "^9.0.0"
  },
  "devDependencies": {
    "@babel/core": "^7.20.0",
    "@babel/preset-env": "^7.20.0",
    "@babel/runtime": "^7.20.0",
    "@react-native/eslint-config": "^0.72.2",
    "@react-native/metro-config": "^0.72.11",
    "@tsconfig/react-native": "^3.0.0",
    "@types/lodash.flatten": "^4.4.6",
    "@types/lodash.merge": "^4.6.6",
    "@types/lodash.startcase": "^4.4.7",
    "@types/react": "^18.0.24",
    "@types/react-native": "0.71.5",
    "babel-plugin-module-resolver": "^5.0.0",
    "eslint": "^8.48.0",
    "eslint-import-resolver-typescript": "^2.5.0",
    "metro-react-native-babel-preset": "^0.77.0",
    "pod-install": "0.1.38",
    "prettier": "^2.8.4",
    "react-native-svg-transformer": "^0.14.3",
    "typescript": "^5.0.4"
  },
  "engines": {
    "node": ">=16"
  },
  "installConfig": {
    "hoistingLimits": "workspaces"
  }
}
