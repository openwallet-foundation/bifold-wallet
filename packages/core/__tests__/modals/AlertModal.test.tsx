import { render } from '@testing-library/react-native'
import React from 'react'

import AlertModal from '../../src/components/modals/AlertModal'
import { BasicAppContext } from '../helpers/app'

describe('AlertModal Component', () => {
  test('Renders correctly', async () => {
    const tree = render(
      <BasicAppContext>
        <AlertModal title="Hello" message="World" />
      </BasicAppContext>
    )

    expect(tree).toMatchSnapshot()
  })
})
