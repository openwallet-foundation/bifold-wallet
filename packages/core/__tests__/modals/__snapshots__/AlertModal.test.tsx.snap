// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlertModal Component Renders correctly 1`] = `
<RNCSafeAreaView
  edges={
    Object {
      "bottom": "additive",
      "left": "additive",
      "right": "additive",
      "top": "additive",
    }
  }
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "column",
      "justifyContent": "center",
      "minHeight": 1334,
    }
  }
>
  <Modal
    hardwareAccelerated={false}
    transparent={true}
    visible={true}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "rgba(0, 0, 0, 0.5)",
          "flex": 1,
          "justifyContent": "center",
          "padding": 20,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#313132",
            "borderColor": "#0099FF",
            "borderRadius": 5,
            "borderWidth": 1,
            "minWidth": 700,
            "padding": 10,
          }
        }
      >
        <View
          style={
            Object {
              "flexDirection": "row",
              "paddingHorizontal": 5,
              "paddingTop": 5,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignSelf": "center",
                  "marginRight": 10,
                },
                Object {
                  "flexDirection": "row",
                },
              ]
            }
          >
            <Text
              accessible={false}
              allowFontScaling={false}
              selectable={false}
              style={
                Array [
                  Object {
                    "color": "#0099FF",
                    "fontSize": 30,
                  },
                  undefined,
                  Object {
                    "fontFamily": "Material Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              î¢Ž
            </Text>
            <Text
              maxFontSizeMultiplier={2}
              style={
                Array [
                  Object {
                    "color": "#FFFFFF",
                    "fontSize": 18,
                    "fontWeight": "bold",
                  },
                  Object {
                    "alignSelf": "center",
                    "color": "#FFFFFF",
                    "flexShrink": 1,
                    "marginLeft": 7,
                  },
                ]
              }
              testID="com.ariesbifold:id/HeaderText"
            >
              Hello
            </Text>
          </View>
        </View>
        <RCTScrollView
          style={
            Object {
              "flexDirection": "column",
              "flexGrow": 0,
              "marginLeft": 40,
              "paddingBottom": 5,
              "paddingHorizontal": 5,
            }
          }
        >
          <View>
            <Text
              maxFontSizeMultiplier={2}
              style={
                Array [
                  Object {
                    "color": "#FFFFFF",
                    "fontSize": 18,
                    "fontWeight": "normal",
                  },
                  Object {
                    "color": "#FFFFFF",
                    "flexShrink": 1,
                    "marginVertical": 16,
                  },
                ]
              }
              testID="com.ariesbifold:id/BodyText"
            >
              World
            </Text>
            <View
              style={
                Object {
                  "paddingTop": 10,
                }
              }
            >
              <View
                accessibilityLabel="Global.Okay"
                accessibilityRole="button"
                accessibilityState={
                  Object {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": false,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  Object {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "backgroundColor": "#42803E",
                    "borderRadius": 4,
                    "opacity": 1,
                    "padding": 16,
                  }
                }
                testID="com.ariesbifold:id/Global.Okay"
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "justifyContent": "center",
                    }
                  }
                >
                  <Text
                    maxFontSizeMultiplier={2}
                    style={
                      Array [
                        Object {
                          "color": "#FFFFFF",
                          "fontSize": 18,
                          "fontWeight": "normal",
                        },
                        Array [
                          Object {
                            "color": "#FFFFFF",
                            "fontSize": 18,
                            "fontWeight": "bold",
                            "textAlign": "center",
                          },
                          false,
                          false,
                          false,
                        ],
                      ]
                    }
                  >
                    Global.Okay
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </RCTScrollView>
      </View>
    </View>
  </Modal>
</RNCSafeAreaView>
`;
